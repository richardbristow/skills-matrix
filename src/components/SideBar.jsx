import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import styled from 'styled-components/macro';

const StyledSideBar = styled.aside`
  grid-area: sidebar;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.sidebarBackground};

  ul {
    padding: 0;
    margin-top: 20px;
    list-style-type: none;

    li {
      color: ${({ theme }) => theme.sidebarText};
      &:hover {
        background-color: ${({ theme }) => theme.sidebarHover};
        cursor: pointer;
      }
      .active {
        background-color: ${({ theme }) => theme.sidebarActiveLink};
      }
    }
  }
`;

const StyledSidebarLink = styled(NavLink)`
  text-decoration: none;
  &:focus,
  &:hover,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
    color: ${({ theme }) => theme.sidebarText};
  }
  padding: 20px 20px 20px 40px;
  display: block;
`;

const SideBar = () => {
  return (
    <StyledSideBar>
      <ul>
        <li>
          <StyledSidebarLink to="/" as={Link}>
            <h3>Skills Test</h3>
          </StyledSidebarLink>
        </li>
        <li>
          <StyledSidebarLink to="/login">Login</StyledSidebarLink>
        </li>
      </ul>
    </StyledSideBar>
  );
};

export default SideBar;
