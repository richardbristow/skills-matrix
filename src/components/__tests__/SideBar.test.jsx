import React from 'react';
import { shallow } from 'enzyme';
import SideBar from '../SideBar';

describe('SideBar', () => {
  it('should render SideBar without any errors', () => {
    const wrapper = shallow(<SideBar />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
