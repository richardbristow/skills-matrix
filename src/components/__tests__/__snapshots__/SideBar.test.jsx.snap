// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should render SideBar without any errors 1`] = `
<div>
  <aside
    class="SideBar__StyledSideBar-sc-1l67eb2-0 lldNPQ"
  >
    <ul
      class="SideBar__StyledSideBarMenu-sc-1l67eb2-1 lbJvnL"
    >
      <li
        class="SideBar__StyledSidebarMenuLink-sc-1l67eb2-3 bEKnlD"
      >
        <a
          class="SideBar__StyledSideBarLink-sc-1l67eb2-2 hQCXjR"
          header="true"
          href="/"
        >
          <h3>
            Skills Matrix
          </h3>
        </a>
      </li>
      <li
        class="SideBar__StyledSidebarMenuLink-sc-1l67eb2-3 ysUGI"
      >
        <a
          class="SideBar__StyledSideBarLink-sc-1l67eb2-2 hQCXjR"
          header="false"
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </aside>
</div>
`;
